storybookjs/storybook
# Release: New release tooling

Works on #22532. This PR is too big to summarize, but it basically implements the whole plan laid out in the RFC for release tooling: https://github.com/storybookjs/storybook/discussions/22182



## What I did

1. Add `scripts/release` that are Node CLI scripts to support the new release workflow
2. Add GitHub Actions that orchestrate those scripts to automate the release process
3. Minor small changes here and there to make the scripts work



## How to test

My initial plan of attack here:

1. get the last parts in from Kasper when they are done
2. Merge this PR
4. See if it works - it should open a pull request
5. Make any necessary changes to make it actually work (maybe we need to add a git tag, or something similar like that)
6. Merge the release PR (to trigger a new alpha release), which should fail because there is no `NPM_TOKEN` in GitHub secrets yet.
7. Inspect the publish workflow run and make sure that it would actually have done the correct thing if it could
8. If everything is good, add the `NPM_TOKEN` secret and rerun the publish workflow.
9. 🤞 see it publish a new alpha version successfully.

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
